### ISSUE TITLE
[Feature]: Deprecate Theme `pull`/`push` commands in favor of `download`/`upload`

### ISSUE THREAD

**Issue by karreiro** [Shopify/shopify-cli#2045]
### What type of project(s) will this request affect?

theme

### What type of change do you want to see?

Substantial change to existing feature

### Overview

Deprecate `shopify theme pull` and `shopify theme push` in favor of, respectively, `shopify theme download` and `shopify theme upload`.

⚠️ Also, consider the same transition for `extension` and `script` projects.

### Motivation

From the DX perspective, `pull`/`push` are inevitably associated with Git `pull`/`push` commands, which carry more complex features, meaning, and expectations.

To precisely express the features of Theme commands and avoid breaking partners' expectations, this issue proposes the renaming Theme `pull`‌/`push` commands to `download`‌/`upload`.

**Comment by binarymonkey84** [comment]
Would agree with the renaming of these tasks. It's confusing for those new to the CLI especially when they are working with the Shopify GitHub integration as well - it's almost implying some kind of link between then since `push` and `pull` are associated with Git nomenclature.

**Comment by karreiro** [comment]
Hi @ADTC :) Would you have any thoughts about this feature?

It's inspired by #1682. As it mentions, the current commands `pull`/`push` are not Git-aware, so I believe renaming them to reflect their actual purpose seems like a promising first step.

Still, `pull`/`push` is widely used by developers.

Just adding a bit more context. For supporting #1682, I believe we will need to find a balance between:
- the Git `pull`/`push` features
- the Shopify (theme) CLI features
- the GitHub Integration features (optional, for users that opt-in)

In other words, some developers may not use Git, so they could rely on `theme upload`. Still, other developers that use Git (hopefully) could rely on `theme push` for a more Git-aware upload.

I'm not proposing that it needs work like this (as we can discuss more at #1682), but I'm exemplifying how freeing up those commands would allow us to grow in that direction :)

**Comment by lesterdefreitas** [comment]
As long as we can maybe have some shorthand versions of those commands as it's longer to type. 😁 

`shopify theme download` > ` shopify theme dl`
`shopify theme upload` > ` shopify theme ul`

Or something like that.

Also, making this change will cause a LOT of general confusion for a while, I'm sure of it.

**Comment by lemonmade** [comment]
cc/ @vividviolet @t6d for thoughts on how `upload` feels as a replacement for `extension push` (I don't think extensions have the equivalent of `pull`)

**Comment by karreiro** [comment]
Many people we've reached out to speak about this proposal agree with renaming `pull`/`push` commands; however, we are far from unanimity.

Given the impact of this change on developers' workflow, and the small sample space we have, I've decided to close this issue for now.

Please leave a message if you feel that `pull`/`push` commands **should** be renamed in favor of `download`/`upload` to provide a better DX! :)

**Comment by ADTC** [comment]
Thank you for closing it! I'm in favor of keeping it as `push` and `pull` for various reasons.
